src/Corax/IndexWriter.cs

```diff
@@ -361,23 +436,75 @@ public long Index(string id, Span<byte> data)
         }
 
-        public unsafe long Index(string id, Span<byte> data, float documentBoost)
+        public long Index(string id, Span<byte> data, float documentBoost)
         {
             var entryId = Index(id, data);
             AppendDocumentBoost(entryId, documentBoost);
             return entryId;
         }
+        
+        private unsafe long Index(Slice id, Span<byte> data)
+        {
+            _numberOfModifications++;
+            Span<byte> buf = stackalloc byte[10];
+            var idLen = ZigZagEncoding.Encode(buf, id.Size);
+            int requiredSize = idLen + id.Size + data.Length;
+            // align to 16 bytes boundary to ensure that we have some (small) space for updating in-place entries
+            requiredSize += 16 - (requiredSize % 16);
+            var entryId = Container.Allocate(Transaction.LowLevelTransaction, _entriesContainerId, requiredSize, out var space);
+            buf.Slice(0, idLen).CopyTo(space);
+            space = space.Slice(idLen);
+            id.CopyTo(space);
+            space = space.Slice(id.Size);
+            data.CopyTo(space);
+            space = space.Slice(data.Length);
+            space.Clear();// clean any old data that may have already been there
+
+            var context = Transaction.Allocator;
+
+            fixed (byte* newEntryDataPtr = data)
+            {
+                var entryReader = new IndexEntryReader(newEntryDataPtr, data.Length);
+
+                foreach (var binding in _fieldsMapping)
+                {
+                    if (binding.FieldIndexingMode is FieldIndexingMode.No)
+                        continue;
+
+                    var indexer = new TermIndexer(this, context, entryReader.GetFieldReaderFor(binding.FieldId), _knownFieldsTerms[binding.FieldId], entryId);
+                    indexer.InsertToken();
+                }
+
+                var it = new IndexEntryReader.DynamicFieldEnumerator(entryReader);
+                while (it.MoveNext())
+                {
+                    var fieldReader = entryReader.GetFieldReaderFor(it.CurrentFieldName);
+
+                    var indexedField = GetDynamicIndexedField(context, ref it);
+
+                    if (indexedField.FieldIndexingMode is FieldIndexingMode.No)
+                        continue;
+
+
+                    var indexer = new TermIndexer(this, context, fieldReader, indexedField, entryId);
+                    indexer.InsertToken();
+                }
+
+                return EntryIdEncodings.Encode(entryId, 1, TermIdMask.Single);
+            }
+        }
 
         //Document Boost should add priority to some documents but also should not be the main component of boosting.
         //The natural logarithm slows down our scoring increase for a document so that the ranking calculated at query time is not forgotten.
+        //We've to add entry container id (without frequency etc) here because in 'SortingMatch' we have already decoded ids.
```

Comment:
And isn't that the _entry id_ here? Why do we have frequency here in the first place.
That is when we do that in the terms entries, unless I missed something

Reply:
Updating document boost happens after indexing the terms.  I didn't wanted to create multiple schemas. It's very convenient to do everything with the same pattern.

I discard the frequency since it's convenient for us to do lookups during query:
https://github.com/ravendb/ravendb/blob/b1b323d777173fcfdc0753de6fbd02054230cdbf/src/Corax/Queries/SortingMatch/SortingMatch.cs#L235-L251